### PR TITLE
Add cosmos transaction definitions

### PR DESCRIPTION
**Pull request by patiee**

### CODE REVIEW

**Comment by jason-c-child** [comment]
This PR was opened 14 days ago. I do not see any comments or activity (review or otherwise). It looks like there's a merge conflict rn.

**Comment by pienkowb** [comment]
@jason-c-child This is a part of Cosmos transaction decoding which is still WIP.

**Comment by patiee** [comment]
Do we want to keep all Tendermint-like networks under one directory or leave it under `chain`?

File: chain/tendermint/cosmos.ts [hunk 1]
```diff
@@ -0,0 +1,4786 @@
...
+
+    export function decodeMsgConnectionOpenAck(m: Uint8Array): MsgConnectionOpenAck {
+      return Protobuf.decode<MsgConnectionOpenAck>(m, MsgConnectionOpenAck.decode)
+    }
+
+    export function decodeMsgConnectionOpenConfirm(
+      m: Uint8Array,
+    ): MsgConnectionOpenConfirm {
+      return Protobuf.decode<MsgConnectionOpenConfirm>(m, MsgConnectionOpenConfirm.decode)
+    }
```
Comment (pienkowb) [inline chain/tendermint/cosmos.ts:4784]: Are these methods generated too or did you add them manually?

File: chain/tendermint/tendermint.ts [hunk 2]
```diff
@@ -18,9 +18,9 @@ export namespace tendermint {
 
   export class EventData {
     public event: Event
-    public block: EventList
+    public block: EventBlock
```
Comment (lavabyrd) [inline chain/tendermint/tendermint.ts:21]: The current main/master already has this fix, rebase? https://github.com/graphprotocol/graph-ts/blob/main/chain/tendermint.ts#L20

File: chain/tendermint/cosmos.ts (hunk 1 above)
Comment (patiee) [reply chain/tendermint/cosmos.ts:4784]: these methods are generated, I showed you the script yesterday. It's in closed PR in graph-proto-as-schema

File: chain/tendermint/tendermint.ts (hunk 2 above)
Comment (patiee) [reply chain/tendermint/tendermint.ts:21]: oh yes, need to merge main branch.

File: yarn.lock [hunk 3]
```diff
@@ -2,24 +2,30 @@
...
 
+as-proto@^v0.2.0:
+  version "0.2.0"
+  resolved "https://registry.yarnpkg.com/as-proto/-/as-proto-0.2.0.tgz#bef36378b373106238d2d03d26dd31e72f423ffd"
+  integrity sha512-uxOCozku/ZcWrqNDyBI4mkkHGcFe522Ev1ZeUFBOxXvkto85saKE0K034eR18ioG6NqNWB7IaXfbfZQmhKpTWg==
+
 assemblyscript@0.19.10:
   version "0.19.10"
-  resolved "https://registry.yarnpkg.com/assemblyscript/-/assemblyscript-0.19.10.tgz#7ede6d99c797a219beb4fa4614c3eab9e6343c8e"
+  resolved "https://registry.npmjs.org/assemblyscript/-/assemblyscript-0.19.10.tgz"
```
Comment (lavabyrd) [inline yarn.lock:12]: any idea why these switched to npmjs instead of yarnpkg? Can we switch them back?

**Review by lavabyrd — CHANGES_REQUESTED** [review]
just the one point on the lock file.

Otherwise my thinking is I'd rather we have a "highlevel" tendermint.ts file that imports each chain specific piece as required

so 
chain
  tendermint.ts
  tendermint_chains/
    cosmos.ts
    terra.ts

etc we could always look at this later though, its not a show stopper for me.

File: yarn.lock (hunk 3 above)
Comment (patiee) [reply yarn.lock:12]: oh good catch thanks, it's probably because I removed and regenerated that file.

**Comment by patiee** [comment]
> just the one point on the lock file.
> 
> Otherwise my thinking is I'd rather we have a "highlevel" tendermint.ts file that imports each chain specific piece as required
> 
> so chain tendermint.ts tendermint_chains/ cosmos.ts terra.ts
> 
> etc we could always look at this later though, its not a show stopper for me.

Thanks Mark for comments,
I will leave Tendermint definitions inside `chain` directory and network-related definitions will be under `tendermint`.

**Comment by sosedoff** [comment]
Let's rebase from main (just did upstream sync). 
As for merging - i've tried running scripts (yarn lint/build/test) and some of them are failing, probably need some small tweaks. 
LGTM if we dont need any more changes.

File: package.json [hunk 4]
```diff
@@ -1,18 +1,20 @@
 {
   "name": "@graphprotocol/graph-ts",
   "description": "TypeScript/AssemblyScript library for writing subgraph mappings for The Graph",
-  "version": "0.24.1",
+  "version": "0.26.0",
```
Comment (lavabyrd) [inline package.json:4]: are we meant to have updated this?

**Comment by lavabyrd** [comment]
also, the package lock.json file is being added. is there a reason we're using that now instead of just the lock file?

**Review by lavabyrd — APPROVED** [review]

**Pull request merged**